substrate/client/service/src/lib.rs

```diff
@@ -372,7 +382,7 @@ pub fn start_rpc_servers<R>(
 	config: &Configuration,
 	gen_rpc_module: R,
 	rpc_id_provider: Option<Box<dyn RpcSubscriptionIdProvider>>,
-) -> Result<Box<dyn std::any::Any + Send + Sync>, error::Error>
+) -> Result<(Box<dyn std::any::Any + Send + Sync>, Option<SocketAddr>), error::Error>
```

Comment:
I'm fine with this.

But the return type is bit complicated and I don't remember why it's `Box<dyn Any>` but would be much nicer with a wrapper type that has an API to get the listen_addr instead returning a tuple.... because the Server type already has this information.

Reply:
I just used the Server type here, and moved the Drop wrapper up to the point where it was actually needed. Let me know if that suffices.